Content on how to run Python

jukent opened the pull request:
This content has information on how to select a Python platform and the basic knowledge needed to install and execute Python code on that platform.

dopplershift commented on foundations/terminal.md:
````diff
@@ -0,0 +1,66 @@
+   $ conda activate pythia_foundations
+   ```
+
+2. Create a directory to store our work. Let's call it `pythia-foundations`.
````
In our workshops, we had issues with novice users getting the directory with their stuff confused with the python environment, so we renamed one to make it easier for them to keep them as separate concepts.

jukent replied:
I think it's a subtle difference (one has an underscore and one a backslash) - I'll make it more obvious. We had the same concern in the Xdev seminar series.

jukent replied:
It is now `pythia_foundations_env` for the environment and `pythia-foundations` for the directory.

jukent commented:
I understand if we want to wait to merge this to see if it should be altered to match the upcoming page template. I also think it could be reasonable for the spin-up content to take on a slightly different form than the package specific pages.

ktyle commented:
Very nice, @jukent . Before I post some specific edits and comments, I'm wondering if for text-centric content, such as this specific document, there might be an easier way than Github pull requests and forks to contribute stylistic changes as well as more content-related edits. I think I'll bring this up at tomorrow's IWG session.

jukent commented:
> Very nice, @jukent . Before I post some specific edits and comments, I'm wondering if for text-centric content, such as this specific document, there might be an easier way than Github pull requests and forks to contribute stylistic changes as well as more content-related edits. I think I'll bring this up at tomorrow's IWG session.

I think that's a great topic for tomorrow.

jukent commented:
I have linked this to hackmd, but am not yet sure how to make it appear on Github.

The links in this PR are:
https://hackmd.io/@jukent/ByFem2ZO_
https://hackmd.io/@jukent/BkHJNnZd_
https://hackmd.io/@jukent/rJlhQ3-_d

clyne commented:
> I have linked this to hackmd, but am not yet sure how to make it appear on Github.
> 
> The links in this PR are:
> https://hackmd.io/LfO88uo7R_C8pypTCH3IWA
> https://hackmd.io/6G271W9GRmWtaNO4chnJUg
> https://hackmd.io/JRLBLL8TSAi7TpOPRcBJpg

The first two links seem to be broken (404). I was able to access the third and successfully fixed a typo and added a test comment. A few questions:

1. Are you able to see the change that I made (the typo that I fixed)?
2. Were you notified that I made changes?
3. Once we're done iterating, how are GitHub and Hackmd synced up?

kmpaul commented:
The three links work for me.  Weird.

jukent commented:
> > I have linked this to hackmd, but am not yet sure how to make it appear on Github.
> > The links in this PR are:
> > https://hackmd.io/LfO88uo7R_C8pypTCH3IWA
> > https://hackmd.io/6G271W9GRmWtaNO4chnJUg
> > https://hackmd.io/JRLBLL8TSAi7TpOPRcBJpg
> 
> The first two links seem to be broken (404). I was able to access the third and successfully fixed a typo and added a test comment. A few questions:
> 
> 1. Are you able to see the change that I made (the typo that I fixed)?
> 2. Were you notified that I made changes?
> 3. Once we're done iterating, how are GitHub and Hackmd synced up?

@clyne can you try again on clicking the links? All the links work for both @kmpaul and myself. In a previous edit the links did not work - so I wonder if it is cached somehow.

1. I was able to see that @kmpaul made changes, but I don't see any changes that you have made. Maybe there is a certain process we have to follow (and document!)
2. I did not receive a notification, I can look at the notification settings.
3. I as the owner of this note and PR fork can go in and push to github with all of the changes we've made included. I think only the owner can do this step.

clyne commented:
> > > I have linked this to hackmd, but am not yet sure how to make it appear on Github.
> > > The links in this PR are:
> > > https://hackmd.io/LfO88uo7R_C8pypTCH3IWA
> > > https://hackmd.io/6G271W9GRmWtaNO4chnJUg
> > > https://hackmd.io/JRLBLL8TSAi7TpOPRcBJpg
> > 
> > 
> > The first two links seem to be broken (404). I was able to access the third and successfully fixed a typo and added a test comment. A few questions:
> > 
> > 1. Are you able to see the change that I made (the typo that I fixed)?
> > 2. Were you notified that I made changes?
> > 3. Once we're done iterating, how are GitHub and Hackmd synced up?
> 
> @clyne can you try again on clicking the links? All the links work for both @kmpaul and myself. In a previous edit the links did not work - so I wonder if it is cached somehow.
> 
> 1. I was able to see that @kmpaul made changes, but I don't see any changes that you have made. Maybe there is a certain process we have to follow (and document!)
> 2. I did not receive a notification, I can look at the notification settings.
> 3. I as the owner of this note and PR fork can go in and push to github with all of the changes we've made included. I think only the owner can do this step.

Well that is interesting. If I click on the links in the email notifier that I received only the last link works. But the links in the PR seem to work just fine. So problem solved. 

Re not seeing my changes: @kmpaul did you do something to save/export your changes? I see a "Versions and GitHub sync" option in the menu, but that doesn't seem to do what I want (it gives me an option to download the file, that's all)

kmpaul commented:
No.  I didn't do anything special to export changes.

clyne commented:
> No. I didn't do anything special to export changes.

Neither of you can see the comment I left at line 13 of the document "Python in the Terminal:

> []leaving a comment

jukent commented:
@clyne Yes I can see it now! And there is a little logo at the top that you made changes 19 hours ago. Perhaps the changes wasn't appearing on my end yet when I checked.

clyne commented:
> @clyne Yes I can see it now! And there is a little logo at the top that you made changes 19 hours ago. Perhaps the changes wasn't appearing on my end yet when I checked.

Maybe just a synchronization issue. Who knows. Glad to know it's working.

brian-rose commented on foundations/jupyter.md:
````diff
@@ -0,0 +1,62 @@
+Then create a Conda environment with Python installed. In the terminal type:
+
+```
+$ conda create --name pythia_foundations_env python jupyterlab nb_conda_kernels
````
I suggest simplifying this so it has a more "quickstart" flavor:

```
conda create --name pythia_foundations_env jupyterlab
```

gives everything you need to start up a Python session in Jlab. By avoiding `nb_conda_kernels`, you get a simpler set of choices when you open up the lab session.

brian-rose commented on foundations/jupyter.md:
```diff
@@ -0,0 +1,62 @@
+
+1. With your Conda environment activated and Jupyter session launched (see above),create a directory to store our work. Let's call it `pythia-foundations`.
+
+   ![Jupyter GUI](https://i.imgur.com/J5l9f9l.png)
```
Update the screenshot if you get rid of `nb_conda_kernels`

jukent replied:
Hmm my Jupyter terminal doesn't look any differently. Perhaps I have nb_conda_kernels installed at base. What is the easiest way to make Jupyter appear as if it were a brand new install?

brian-rose replied:
As we discussed at today's EWG, it's probably as you said because you have `nb_conda_kernels` in base.

I took a new screenshot and dropped it into https://github.com/ProjectPythia/pythia-foundations/pull/48/commits/2240de3ac6aa047d8c254efd225e6b13d088304e

brian-rose commented on foundations/jupyter.md:
```diff
@@ -0,0 +1,62 @@
+
+   ![Jupyter - Hello World](https://i.imgur.com/InDEyPn.png)
+
+**Congratulations!** You have just set up your first Python environment and run your first Python code in a Jupyter notebook.
```
Should we include some instructions for how to save the notebook and shut down the lab session?

jukent replied:
Good idea!

brian-rose replied:
Added a section heading for this in https://github.com/ProjectPythia/pythia-foundations/pull/48/commits/2240de3ac6aa047d8c254efd225e6b13d088304e

brian-rose reviewed (commented):
I made some edits to the text in several documents over on hackmd

brian-rose commented:
Let me try that again, because I don't see the updated screenshot in the rendered version.

brian-rose commented:
Ok, I see that the markdown file didn't actually link to the images in the repo at all but pointed to imgur links. I replaced those so the new screenshot should show up this time.

jukent commented:
> Ok, I see that the markdown file didn't actually link to the images in the repo at all but pointed to imgur links. I replaced those so the new screenshot should show up this time.

The markdown file used the images before, but HackMD doesn't seem to be able to see them - hence the imgur links. This is perhaps another pain point of HackMD.

Thanks for grabbing that screenshot!

brian-rose commented:
Yeah, maybe HackMD is not quite the right tool for this job.

brian-rose commented:
We should close this since all the content has been merged into #50 which is still getting reviewed.

brian-rose commented:
This is out of date now and the new content has been merged in #50. Closing this.
